### PR TITLE
Feat/upstash redis storage

### PR DESCRIPTION
**Pull request by bracesproul**
Added upstash redis storage & tests

### CODE REVIEW

**Comment by langchain-infra** [comment]
https://langchainjs-docs-jr4el4zc3-langchain.vercel.app

File: .vscode/settings.json [hunk 1]
```diff
@@ -3,10 +3,11 @@
     "./langchain",
```
Comment (jacoblee93) [inline .vscode/settings.json:3]: I actually don't think this should be in the codebase at all?

**Comment by jacoblee93** [comment]
Thanks, don't forget to add entrypoints!

**Comment by bracesproul** [comment]
> Thanks, don't forget to add entrypoints!

Shoot, good catch haha 😅

File: .vscode/settings.json (hunk 1 above)
Comment (bracesproul) [reply .vscode/settings.json:3]: [This stack overflow answer](https://stackoverflow.com/questions/32964920/should-i-commit-the-vscode-folder-to-source-control#:~:text=Recommendation%20is%20to%20generally%20exclude,a%20commit%20is%20checked%20out) is saying it's good practice to include a couple, for sharing certain settings across devs.

File: langchain/scripts/create-entrypoints.js [hunk 2]
```diff
@@ -257,6 +257,8 @@ const entrypoints = {
   "storage/in_memory": "storage/in_memory",
   "storage/ioredis": "storage/ioredis",
   "storage/vercel_kv": "storage/vercel_kv",
+  "storage/upstash_redis": "storage/upstash_redis",
```
Comment (jacoblee93) [inline langchain/scripts/create-entrypoints.js:260]: Also needs to go under `requiresOptionalDependency` below since it uses `@upstash/redis`

**Review by jacoblee93 — COMMENTED** [review]
One more thing!

**Comment by jacoblee93** [comment]
Thank you!

**Pull request merged**